components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RegistryContentResource.java

```diff
@@ -0,0 +1,441 @@
+    Set<String> methods;
+
+    public RegistryContentResource() {
+
+        methods = new HashSet<>();
+        methods.add(Constants.HTTP_GET);
+        methods.add(Constants.HTTP_POST);
+        methods.add(Constants.HTTP_PUT);
+        methods.add(Constants.HTTP_DELETE);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext messageContext, org.apache.axis2.context.MessageContext axis2MessageContext,
+            SynapseConfiguration synapseConfiguration) {
+
+        String httpMethod = axis2MessageContext.getProperty(Constants.HTTP_METHOD_PROPERTY).toString();
+        if (httpMethod.equals(HTTP_GET)) {
+            handleGet(messageContext, axis2MessageContext);
+        } else {
+            try {
+                if (SecurityUtils.isAdmin(messageContext.getProperty(USERNAME_PROPERTY).toString())) {
+                    switch (httpMethod) {
+                    case HTTP_POST:
+                        handlePost(messageContext, axis2MessageContext);
+                        break;
+                    case HTTP_PUT:
+                        handlePut(messageContext, axis2MessageContext);
+                        break;
+                    case HTTP_DELETE:
+                        handleDelete(messageContext, axis2MessageContext);
+                        break;
+                    default:
+                        JSONObject jsonBody = Utils.createJsonError("Unsupported HTTP method, " + httpMethod
+                                        + ". Only GET, POST, PUT and DELETE methods are supported", axis2MessageContext,
+                                BAD_REQUEST);
+                        Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+                        axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+                        break;
+                    }
+                } else {
+                    Utils.sendForbiddenFaultResponse(axis2MessageContext);
+                }
+            } catch (UserStoreException e) {
+                JSONObject jsonBody = Utils.createJsonError("Error occurs while retrieving the user data",
+                        axis2MessageContext, INTERNAL_SERVER_ERROR);
+                Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+                axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+            }
+        }
+        return true;
+    }
+
+    /**
+     * This method handles GET.
+     *
+     * @param messageContext      Synapse message context
+     * @param axis2MessageContext AXIS2 message context
+     */
+    private void handleGet(MessageContext messageContext, org.apache.axis2.context.MessageContext axis2MessageContext) {
+
+        String registryPath = Utils.getQueryParameter(messageContext, REGISTRY_PATH);
+        String validatedPath;
+        if (Objects.nonNull(registryPath)) {
```

Comment:
can we parse the registryPath to a string and check for both null and empty string. (since user input)
does checking not null only is enough ?

Reply:
Changed it to check for empty strings as well